### PR TITLE
Fix inefficient XContentParser instances using BytesReference.streamInput()

### PR DESCRIPTION
**Pull request by original-brownbear**
BytesReference are very often if not most of the time array backed. We can use the helper utility in all production spots to save copying every byte twice during parsing.
Did that here and dried up the logic for getting a parser config in a couple spots.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java [hunk 1]
```diff
@@ -436,11 +433,7 @@ public static RoleDescriptor parse(
     ) throws IOException {
         assert name != null;
         // EMPTY is safe here because we never use namedObject
```
Comment (ldematte) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java:435]: Nit: move or remove this comment

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/Pivot.java [hunk 2]
```diff
@@ -210,14 +210,11 @@ private static CompositeAggregationBuilder createCompositeAggregationSources(Piv
...
-                XContentParser parser = builder.generator()
-                    .contentType()
-                    .xContent()
-                    .createParser(
-                        NamedXContentRegistry.EMPTY,
-                        LoggingDeprecationHandler.INSTANCE,
-                        BytesReference.bytes(builder).streamInput()
-                    )
+                XContentParser parser = XContentHelper.createParserNotCompressed(
+                    XContentParserConfiguration.EMPTY,
```
Comment (ldematte) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/Pivot.java:214]: Shouldn't this be `XContentHelper.LOG_DEPRECATIONS_CONFIGURATION`?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/support/xcontent/XContentSource.java [hunk 3]
```diff
@@ -135,7 +132,7 @@ public static void writeTo(XContentSource source, StreamOutput out) throws IOExc
     private Object data() {
         if (data == null) {
             // EMPTY is safe here because we never use namedObject
```
Comment (ldematte) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/support/xcontent/XContentSource.java:134]: Nit: move or remove this comment

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/support/xcontent/XContentSource.java [hunk 4]
```diff
@@ -112,15 +109,15 @@ public <T> T getValue(String path) {
...
+        try (XContentParser parser = parser(bytes)) {
             parser.nextToken();
             builder.generator().copyCurrentStructure(parser);
             return builder;
         }
     }
 
-    public XContentParser parser(NamedXContentRegistry xContentRegistry, InputStream stream) throws IOException {
-        return contentType.xContent().createParser(xContentRegistry, LoggingDeprecationHandler.INSTANCE, stream);
+    public XContentParser parser(BytesReference stream) throws IOException {
```
Comment (ldematte) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/support/xcontent/XContentSource.java:119]: Nit: rename parameter

**Review by ldematte — APPROVED** [review]
Just one comment (plus 3 nits) to address, and be sure the test failure on `RestControllerTests` is not related. Otherwise, LGTM!

**Comment by original-brownbear** [comment]
Thanks Lorenzo! Test failure was related but a trivial fix :) Turns out for byte arrays Jackson validates that the content is actually SMILE for smile parsers at construct time so I forced the random content to still start on the smily.

**Pull request merged**